### PR TITLE
Feature: Embed explorable explanations in PSAT extension

### PR DESCRIPTION
**Pull request by amovar18**
## Description
This PR aims to embed explorable explanation in PSAT extension.

## Relevant Technical Choices
- Webpack in production mode now gives a dist file from Explorable Explanations package.
- This package exports `userSketch`, `interestGroupSketch` and main sketch which is imported by the extension package.

## Testing Instructions
- Clone this branch.
- In the terminal run `npm i && npm run build:ext`
- Now go to example.com and open the DevTools.
- In DevTools navigate to the PSAT extension.
- Go to Protected Audience tab and select explorable explanations you should now see the animation.

## Screenshot/Screencast

https://github.com/user-attachments/assets/b4352d38-1b5c-4e39-ab47-cca911d77871

---

## Checklist

- [ ] I have thoroughly tested this code to the best of my abilities.
- [ ] I have reviewed the code myself before requesting a review.
- [ ] This code is covered by unit tests to verify that it works as intended.
- [ ] The QA of this PR is done by a member of the QA team (to be checked by QA).

Partially addresses #857

### CODE REVIEW

**Comment by mohdsayed** [comment]
@amovar18 

- [ ]  Reduce gap between “next” and “prev” buttons, add title and change opacity on hover.
- [ ]  Center the expanded circle popup and reduce size.
- [ ]  Change the speed of barrage proportional to the distance between the starting and end point.
- [ ]  Interactive Mode: Remove user icon from cursor when it goes outside of the canvas.
- [ ]  IM: Remove next and previous buttons in interactive mode.
- [ ]  In auction, ripple effect covers above down arrow.

For next PRs:

- [ ]  Make reset button work.
- [ ]  Speed should be controlled from the speed slider.

File: packages/explorable-explanations/src/config.js [hunk 1]
```diff
@@ -135,7 +135,13 @@ const config = {
   isInteractiveMode: false,
   mouseX: 0,
   mouseY: 0,
+  nextNodeIndexTracker: 1,
+  prevNodeIndexTracker: 0,
   shouldRespondToClick: true,
+  startTrackingMouse: true,
+  isReset: false,
+  animationFrames: [],
+  wasAnythingDoneInInteractiveMode: false,
```
[the remark this answers is not in the document]
Comment (mohdsayed) [reply packages/explorable-explanations/src/config.js:144]: If these tracking the state somehow, `config` is not the place for it. They should be moved `app`.

File: packages/explorable-explanations/src/lib/utils.js [hunk 2]
```diff
@@ -190,44 +216,46 @@ utils.setupInterestGroupCanvas = (p) => {
   overlayCanvas.parent('interest-canvas');
   overlayCanvas.style('z-index', 2);
   p.textSize(config.canvas.fontSize);
+  // eslint-disable-next-line no-undef
+  if (process.env.NODE_ENV !== 'production') {
```
Comment (mohdsayed) [inline packages/explorable-explanations/src/lib/utils.js:220]: What does this do for dev mode?

File: packages/explorable-explanations/src/modules/auctions.js [hunk 3]
```diff
@@ -320,37 +446,43 @@ auction.draw = async (index) => {
...
-      });
-    }
+        utils.wipeAndRecreateInterestCanvas(); // eslint-disable-line no-await-in-loop
+      }
 
-    if (callBack) {
-      callBack(returnValue);
-    }
-    await utils.delay(delay); // eslint-disable-line no-await-in-loop
+      if (props?.title === 'generateBid()') {
```
Comment (mohdsayed) [inline packages/explorable-explanations/src/modules/auctions.js:464]: As we have to check conditions and text can change in future, let's add keys to the object for comparison.

**Review by mohdsayed — CHANGES_REQUESTED** [review]
We can merge after some improvements.

File: packages/explorable-explanations/src/config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (amovar18) [reply packages/explorable-explanations/src/config.js]: Removed and moved all the state to `app`.

File: packages/explorable-explanations/src/lib/utils.js (hunk 2 above)
Comment (amovar18) [reply packages/explorable-explanations/src/lib/utils.js]: Using a custom set env variable that will decide if canvas is running standalone or with not.

File: packages/explorable-explanations/src/modules/auctions.js (hunk 3 above)
Comment (amovar18) [reply packages/explorable-explanations/src/modules/auctions.js]: Resolved now using `extraProps` object instead of title for comparison.

**Review by mohdsayed — APPROVED** [review]
Looks good, thanks

**Pull request merged**